### PR TITLE
Breathing - Add cyanosis diagnose

### PR DESCRIPTION
**Pull request by MiszczuZPolski**
**When merged this pull request will:**
- Add cyanosise diagnose with 4 settable levels:
1. No Cyanosis - 100% SpO2 (no settable)
2. Slight Cyanosis - over 90% SpO2 (default)
3. Mild Cyanosis - over 75% SpO2 (default)
4. Severe Cyanosis - over 66% SpO2 (default)

Because some groups doesn't use lethal SpO2 value or have it different from default 65%, all levels are adjustable.

To do:

- Translation for more languages ._.

### CODE REVIEW

**Comment by bababoomjat** [comment]
The level of "no cyanosys" could be modified to be at least over ~95%? Or make it settable? 
I reckon that 100% is good for gameplay perspective, but from a immersion perspective a slight cianosis at 99% would mean that 3/4 of world population is cianotic.

https://journals.lww.com/pec-online/Abstract/2015/09000/Pulse_Oxygen_Saturation_Values_in_a_Healthy.7.aspx
This is a shoold-aged population but the concept is similar

And this without accounting for blood loss that could make cianosis even harder to detect, although i guess that such code would be much more difficult to implement

I don't mean to denigrate it, it's a great idea imo beside this detail!

**Comment by MiszczuZPolski** [comment]
> The level of "no cyanosys" could be modified to be at least over ~95%? Or make it settable? I reckon that 100% is good for gameplay perspective, but from a immersion perspective a slight cianosis at 99% would mean that 3/4 of world population is cianotic.

I know IRL normal person would have SpO2 lower than 100%, but for gameplay purpose i want to keep it as 100% - that means when someone is injured (for example hidden pneumothorax) medic would know this. 

After commenting this i will add commit for medical level required so Anyone, Medic or Doctor could only detect it based on setting.

File: addons/breathing/stringtable.xml [hunk 1]
```diff
@@ -752,5 +752,73 @@
             <Italian>Valore di SpO2 sotto il quale causa la perdita di coscienza</Italian>
             <Japanese>SpO2(血中酸素濃度)が設定した値を下回ると気絶し、無意識状態に陥ります。</Japanese>
         </Key>
+        <Key ID="STR_kat_breathing_CheckCyanosis_Name">
+            <English>Check Cyanosis</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:756]:
```suggestion
            <English>Check Cyanosis</English>
            <German>Zyanose überprüfen</German>
```

File: addons/breathing/stringtable.xml [hunk 2]
```diff
@@ -752,5 +752,73 @@
...
+            <Polish>Sprawdź sinicę</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_CheckCyanosis_Progress">
+            <English>Checking Cyanosis</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:761]:
```suggestion
            <English>Checking Cyanosis</English>
            <German>Zyanose prüfen</German>
```

File: addons/breathing/functions/fnc_treatmentAdvanced_CyanosisLocal.sqf [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        _spO2Output = LSTRING(CyanosisStatus_Severe);
+        _messageCyanosis = LLSTRING(CyanosisStatus_Severe);
+    };
+
+};
+
+private _message = format ["%1",_messageCyanosis];
+[_message, 2, _player] call ace_common_fnc_displayTextStructured;
+
+[_target, "quick_view", LSTRING(CheckCyanosis_Log), [[_player] call ace_common_fnc_getName, _spO2Output]] call ace_medical_treatment_fnc_addToLog;
```
Comment (Tomcat-SG) [inline addons/breathing/functions/fnc_treatmentAdvanced_CyanosisLocal.sqf]:
```suggestion
[_target, "quick_view", LSTRING(CheckCyanosis_Log), [_spO2Output]] call ace_medical_treatment_fnc_addToLog;
```

Comment (Tomcat-SG) [inline addons/breathing/functions/fnc_treatmentAdvanced_CyanosisLocal.sqf:49]:
```suggestion
[_target, "quick_view", LSTRING(CheckCyanosis_Log), [_spO2Output]] call ace_medical_treatment_fnc_addToLog;
```

File: addons/breathing/stringtable.xml [hunk 4]
```diff
@@ -752,5 +752,73 @@
...
+            <Polish>Sprawdzanie sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_CheckCyanosis_Log">
+            <English>Cyanosis Status: %2</English>
+            <Polish>Status sinicy: %2</Polish>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml]:
```suggestion
            <English>Cyanosis Status: %1</English>
            <German>Status der Zyanose: %1</German>
            <Polish>Status sinicy: %1</Polish>
```

Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:765]:
```suggestion
            <English>Cyanosis Status: %1</English>
            <German>Status der Zyanose: %1</German>
            <Polish>Status sinicy: %1</Polish>
```

File: addons/breathing/stringtable.xml [hunk 5]
```diff
@@ -752,5 +752,73 @@
...
+        </Key>
+        <Key ID="STR_kat_breathing_CyanosisStatus_N">
+            <English>No Cyanosis</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:771]:
```suggestion
            <English>No Cyanosis</English>
            <German>Keine Zyanose</German>
```

Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:768]:
```suggestion
            <English>No Cyanosis</English>
            <German>Keine Zyanose</German>
```

File: addons/breathing/stringtable.xml [hunk 6]
```diff
@@ -752,5 +752,73 @@
...
+            <Polish>Brak sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_CyanosisStatus_Slight">
+            <English>Slight Cyanosis</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:776]:
```suggestion
            <English>Slight Cyanosis</English>
            <German>Leichte Zyanose</German>
```

Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:772]:
```suggestion
            <English>Slight Cyanosis</English>
            <German>Leichte Zyanose</German>
```

File: addons/breathing/stringtable.xml [hunk 7]
```diff
@@ -752,5 +752,73 @@
...
+            <Polish>Lekka sinica</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_CyanosisStatus_Mild">
+            <English>Mild Cyanosis</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:781]:
```suggestion
            <English>Mild Cyanosis</English>
            <German>Mittelschwere Zyanose</German>
```

Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:776]:
```suggestion
            <English>Mild Cyanosis</English>
            <German>Mittelschwere Zyanose</German>
```

File: addons/breathing/stringtable.xml [hunk 8]
```diff
@@ -752,5 +752,73 @@
...
+            <Polish>Umiarkowana sinica</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_CyanosisStatus_Severe">
+            <English>Severe Cyanosis</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:786]:
```suggestion
            <English>Severe Cyanosis</English>
            <German>Schwere Zyanose</German>
```

Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:780]:
```suggestion
            <English>Severe Cyanosis</English>
            <German>Schwere Zyanose</German>
```

File: addons/breathing/stringtable.xml [hunk 9]
```diff
@@ -752,5 +752,73 @@
...
+            <Polish>Ciężka sinica</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_SETTING_Cyanosis">
+            <English>Enable cyanosis diagnose</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:791]:
```suggestion
            <English>Enable cyanosis diagnose</English>
            <German>Zyanose-Diagnose aktivieren</German>
```

Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:784]:
```suggestion
            <English>Enable cyanosis diagnose</English>
            <German>Zyanose-Diagnose aktivieren</German>
```

File: addons/breathing/stringtable.xml [hunk 10]
```diff
@@ -752,5 +752,73 @@
...
+            <Polish>Włącz diagnozę sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_SETTING_Cyanosis_DESC">
+            <English>Enables cyanosis diagnose</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:796]:
```suggestion
            <English>Enables cyanosis diagnose</English>
            <German>Aktiviert die Zyanose-Diagnose</German>
```

Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:788]:
```suggestion
            <English>Enables cyanosis diagnose</English>
            <German>Aktiviert die Zyanose-Diagnose</German>
```

File: addons/breathing/stringtable.xml [hunk 11]
```diff
@@ -752,5 +752,73 @@
...
+            <Polish>Włącza diagnozę sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_SETTING_slightValue">
+            <English>Slight cyanosis SpO2 value</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:801]:
```suggestion
            <English>Slight cyanosis SpO2 value</English>
            <German>Leichter Zyanose SpO2-Wert</German>
```

Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:792]:
```suggestion
            <English>Slight cyanosis SpO2 value</English>
            <German>Leichter Zyanose SpO2-Wert</German>
```

File: addons/breathing/stringtable.xml [hunk 12]
```diff
@@ -752,5 +752,73 @@
...
+            <Polish>Poziom SpO2 dla lekkiej sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_SETTING_slightValue_DESC">
+            <English>Slight cyanosis value - default 90 SpO2</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:806]:
```suggestion
            <English>Slight cyanosis value - default 90 SpO2</English>
            <German>Leichte Zyanose - Standardwert 90 SpO2</German>
```

Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:796]:
```suggestion
            <English>Slight cyanosis value - default 90 SpO2</English>
            <German>Leichte Zyanose - Standardwert 90 SpO2</German>
```

File: addons/breathing/stringtable.xml [hunk 13]
```diff
@@ -752,5 +752,73 @@
...
+            <Polish>Poziom SpO2 dla lekkiej sinicy - domyślnie 90 SpO2</Polish>
+        </Key>
+         <Key ID="STR_kat_breathing_SETTING_mildValue">
+            <English>Mild cyanosis SpO2 value</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:811]:
```suggestion
            <English>Mild cyanosis SpO2 value</English>
            <German>Mittelschwere Zyanose SpO2-Wert</German>
```

Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:800]:
```suggestion
            <English>Mild cyanosis SpO2 value</English>
            <German>Mittelschwere Zyanose SpO2-Wert</German>
```

File: addons/breathing/stringtable.xml [hunk 14]
```diff
@@ -752,5 +752,73 @@
...
+            <Polish>Poziom SpO2 dla umiarkowanej sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_SETTING_mildValue_DESC">
+            <English>Mild cyanosis value - default 75 SpO2</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:816]:
```suggestion
            <English>Mild cyanosis value - default 75 SpO2</English>
            <German>Mittelschwere Zyanose - Standardwert 75 SpO2</German>
```

Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:804]:
```suggestion
            <English>Mild cyanosis value - default 75 SpO2</English>
            <German>Mittelschwere Zyanose - Standardwert 75 SpO2</German>
```

File: addons/breathing/stringtable.xml [hunk 15]
```diff
@@ -752,5 +752,73 @@
...
+            <Polish>Poziom SpO2 dla umiarkowanej sinicy - domyślnie 75 SpO2</Polish>
+        </Key>
+         <Key ID="STR_kat_breathing_SETTING_severeValue">
+            <English>Severe cyanosis SpO2 value</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:821]:
```suggestion
            <English>Severe cyanosis SpO2 value</English>
            <German>Schwere Zyanose SpO2-Wert</German>
```

Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:808]:
```suggestion
            <English>Severe cyanosis SpO2 value</English>
            <German>Schwere Zyanose SpO2-Wert</German>
```

File: addons/breathing/stringtable.xml [hunk 16]
```diff
@@ -752,5 +752,73 @@
...
+            <Polish>Poziom SpO2 dla ciężkiej sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_SETTING_severeValue_DESC">
+            <English>Severe cyanosis value - default 66 SpO2 \nShould be little bit higher than lethal SpO2 value</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:826]:
```suggestion
            <English>Severe cyanosis value - default 66 SpO2 \nShould be little bit higher than lethal SpO2 value</English>
            <German>Schwere Zyanose - Standardwert 66 SpO2 \nSollte etwas höher sein als der tödliche SpO2-Wert</German>
```

Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:812]:
```suggestion
            <English>Severe cyanosis value - default 66 SpO2 \nShould be little bit higher than lethal SpO2 value</English>
            <German>Schwere Zyanose - Standardwert 66 SpO2 \nSollte etwas höher sein als der tödliche SpO2-Wert</German>
```

File: addons/breathing/stringtable.xml [hunk 17]
```diff
@@ -752,5 +752,73 @@
...
+            <Polish>Poziom SpO2 dla ciężkiej sinicy - domyślnie 66 SpO2 \nPowinien być minimalnie wyższy niż wartość śmiertelna SpO2</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_CYANOSIS_TREATMENT_LEVEL">
+            <English>Check Cyanosis Minimum Level</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:831]:
```suggestion
            <English>Check Cyanosis Minimum Level</English>
            <German>Zyanose prüfen Medic Level</German>
```

Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:816]:
```suggestion
            <English>Check Cyanosis Minimum Level</English>
            <German>Zyanose prüfen Medic Level</German>
```

File: addons/breathing/stringtable.xml [hunk 18]
```diff
@@ -752,5 +752,73 @@
...
+            <Polish>Minimalny poziom wyszkolenia do sprawdzenia sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_CYANOSIS_TREATMENT_LEVEL_DESCRIPTION">
+            <English>The minimum medical level required for someone to be able to check cyanosis.</English>
```
Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:836]:
```suggestion
            <English>The minimum medical level required for someone to be able to check cyanosis.</English>
            <German>Das Medic Level, das erforderlich ist, damit jemand eine Zyanose feststellen kann.</German>
```

Comment (Tomcat-SG) [inline addons/breathing/stringtable.xml:820]:
```suggestion
            <English>The minimum medical level required for someone to be able to check cyanosis.</English>
            <German>Das Medic Level, das erforderlich ist, damit jemand eine Zyanose feststellen kann.</German>
```

**Review by Tomcat-SG — CHANGES_REQUESTED** [review]
German translation and argument for log changed

**Review by Tomcat-SG — APPROVED** [review]
LGTM

**Pull request merged**